apisix/admin/standalone.lua

```diff
@@ -0,0 +1,197 @@
+        if req_body[key] and #req_body[key] > 0 then
+            apisix_yaml[key] = table_new(1, 0)
+            local item_schema = obj.item_schema
+            local item_checker = obj.checker
+
+            for index, item in ipairs(req_body[key]) do
+                local valid, err
+                -- need to recover to 0-based subscript
+                local err_prefix = "invalid " .. key .. " at index " .. (index - 1) .. ", err: "
+                if item_schema then
+                    valid, err = check_schema(obj.item_schema, item)
+                    if not valid then
+                        core.log.error(err_prefix, err)
+                        core.response.exit(400, {error_msg = err_prefix .. err,
+                                                req_body = req_body})
+                    end
+                end
+                if item_checker then
+                    valid, err = item_checker(item)
+                    if not valid then
+                        core.log.error(err_prefix, err)
+                        core.response.exit(400, {error_msg = err_prefix .. err,
+                                                req_body = req_body})
+                    end
+                end
+                table_insert(apisix_yaml[key], item)
+            end
+        end
+    end
+
+    local success, err = events:post(EVENT_UPDATE, EVENT_UPDATE, core.json.encode({
+        config = apisix_yaml,
+        conf_version = conf_version,
+    }))
+    if not success then
+        core.response.exit(500, err)
+    end
+
+    core.response.set_header("X-APISIX-Conf-Version", tostring(conf_version))
+    return core.response.exit(202)
+end
+
+
+local function get(ctx)
+    local accept = core.request.header(nil, "accept") or "application/json"
+    local want_yaml_resp = core.string.has_prefix(accept, "application/yaml")
+
+    local _, ver, config = config_yaml._get_config()
+
+    core.response.set_header("X-APISIX-Conf-Version", tostring(ver))
+    local resp, err
+    if want_yaml_resp then
+        core.response.set_header("Content-Type", "application/yaml")
+        resp = yaml.dump({ config })
+        if not resp then
+            err = "failed to encode yaml"
+        end
+
+        -- remove the first line "---" and the last line "..."
+        -- because the yaml.dump() will add them for multiple documents
+        local m = ngx.re.match(resp, [[^---\s*([\s\S]*?)\s*\.\.\.\s*$]], "jo")
+        if m and m[1] then
+            resp = m[1]
+        end
+    else
+        core.response.set_header("Content-Type", "application/json")
+        resp, err = core.json.encode(config, true)
+        if not resp then
+            err = "failed to encode json: " .. err
+        end
+    end
+
+    if not resp then
+        return core.response.exit(500, {error_msg = err})
+    end
+    return core.response.exit(200, resp)
+end
+
+
+function _M.run()
+    local ctx = ngx.ctx.api_ctx
+    if str_lower(get_method()) == "put" then
```

Comment:
minor code style

```lua
if xx == put then
    ... ...
end

if xxx == get the
     return get(ctx)
end

error(wrong method)

Reply:
The method is moved to a separate variable, except that determining `method == X` is not necessary, because only `PUT` and `GET` requests are allowed when this route is created, so there is no need to handle additional unintended methods.

Reply:
code style fixed